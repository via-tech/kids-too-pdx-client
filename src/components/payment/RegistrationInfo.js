import React from 'react';
import PropTypes from 'prop-types';
import styles from './Registration.css';

function RegistrationInfo({ handleChange }) {
  return (
    <fieldset>
      <legend>Organization Info</legend>

      <label>Username<span className={styles.required}>*</span>
        <input type="text" name="username" onChange={handleChange} required />
      </label>

      <label>Organization Name<span className={styles.required}>*</span>
        <input type="text" name="name" onChange={handleChange} required/>
      </label>

      <label>Email<span className={styles.required}>*</span>
        <input type="email" name="email" onChange={handleChange} required />
      </label>

      <label>Phone<span className={styles.required}>*</span>
        <input type="tel" name="phone" onChange={handleChange} required />
      </label>

      <label>Address<span className={styles.required}>*</span>
        <label>Street<span className={styles.required}>*</span>
          <input type="text" name="street" onChange={handleChange} required />
        </label>

        <label>City<span className={styles.required}>*</span>
          <input type="text" name="city" onChange={handleChange} required />
        </label>

        <label>Zip Code<span className={styles.required}>*</span>
          <input type="text" name="zipcode" pattern="[0-9]*" onChange={handleChange} required />
        </label>
      </label>

      <label>Organization Bio:
        <textarea name="bio" onChange={handleChange} />
      </label>

      <label>Password<span className={styles.required}>*</span>
        <input type="password" name="password" onChange={handleChange} required />
      </label>

      <label>Confirm Password<span className={styles.required}>*</span>
        <input type="password" name="confirmPassword" onChange={handleChange} required />
      </label>
    </fieldset>
  );
}

RegistrationInfo.propTypes = {
  handleChange: PropTypes.func
};

export default RegistrationInfo;
