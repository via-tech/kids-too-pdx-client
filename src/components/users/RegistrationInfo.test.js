import React from 'react';
import renderer from 'react-test-renderer';
import RegistrationInfo from './RegistrationInfo';

describe('Registration Form', () => {
  it('snapshots RegistrationInfo form', () => {
    const props = {
      handleChange: jest.fn()
    };

    const tree = renderer.create(
      <RegistrationInfo {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
