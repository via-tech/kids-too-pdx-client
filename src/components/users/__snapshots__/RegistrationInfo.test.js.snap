// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form snapshots RegistrationInfo form 1`] = `
<fieldset>
  <legend>
    Organization Info
  </legend>
  <label>
    Username:
    <input
      name="username"
      onChange={[MockFunction]}
      required={true}
      type="text"
    />
  </label>
  <label>
    Organization Name:
    <input
      name="name"
      onChange={[MockFunction]}
      required={true}
      type="text"
    />
  </label>
  <label>
    Email:
    <input
      name="email"
      onChange={[MockFunction]}
      required={true}
      type="email"
    />
  </label>
  <label>
    Phone:
    <input
      name="phone"
      onChange={[MockFunction]}
      required={true}
      type="tel"
    />
  </label>
  <label>
    Address:
    <label>
      Street:
      <input
        name="street"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      City:
      <input
        name="city"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Zip Code:
      <input
        name="zipcode"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        required={true}
        type="text"
      />
    </label>
  </label>
  <label>
    Organization Bio:
    <textarea
      name="bio"
      onChange={[MockFunction]}
    />
  </label>
  <label>
    Password:
    <input
      name="password"
      onChange={[MockFunction]}
      required={true}
      type="text"
    />
  </label>
</fieldset>
`;
